#  Surréservation et profit p363 **correction**


vOTRE NOM
TELECHARGEZ PUIS POSTEZ A TINYURL.COM/MRCHRETIEN 

L’avion considéré peut contenir 124 passagers. Chaque passager se présente à l’embarquement, indépendamment des autres passagers, avec une probabilité égale à 0,91. Le nombre de billets vendus par la compagnie aérienne est noté n. Cet entier est supérieur ou égal à 124.

L’entreprise cherche ici à optimiser ses profits. Chaque client qui se présente et qui peut embarquer rapporte 100 € à la compagnie aérienne. En revanche, chaque client qui se présente mais qui se voit refuser la possibilité de monter à bord coûte 150 € à la compagnie aérienne. Un client qui ne se présente pas peut se faire rembourser en partie son billet et permet un profit de 30 € à la compagnie aérienne.

1. compléter les fonctions avionProfit et simulations.


In [1]:
from random import random
 
def passagerPresent():
    # Renvoie 1 si le passager se présente à l'embarquement et 0 sinon
    if random() <= 0.91:
        return 1
    else:
        return 0

def avionProfit(n):
    # Renvoie le profit pour un avion, pour n billets vendus
    profit = 0
    passagers = 0
    for passager in range(n):
        if not(passagerPresent()):
            profit = profit + 30
        else:
            if (passagers < 124):
                profit = profit + 100
                passagers = passagers + 1
            else:
                profit = profit - 150
    return profit

def simulations(nbSimulations, n):
    # Renvoie le profit moyen pour nbSimulations avions avec chacun n billets vendus
    profitMoyen = 0
    for avion in range(nbSimulations):
        profitMoyen = profitMoyen + avionProfit(n)
    return profitMoyen/nbSimulations
 
def optimisation():
    # Nombre optimal de billets à vendre pour optimiser le profit moyen par avion
    nombreBilletsVendus = 100
    nombreOptimalBilletsVendus = 100
    profit = 0
    profitOptimal = 0
    while (profit == profitOptimal and nombreBilletsVendus <= 248):
        nombreBilletsVendus = nombreBilletsVendus + 1
        profit = simulations(1000, nombreBilletsVendus)
        if (profit > profitOptimal):
            nombreOptimalBilletsVendus = nombreBilletsVendus
            profitOptimal = profit
    return nombreOptimalBilletsVendus

print("Estimation du profit moyen par avion lorsque n = 130 : ", avionProfit(130))

print("Estimation du nombre de billets à vendre pour maximiser le profit : ", optimisation())


Estimation du profit moyen par avion lorsque n = 130 :  12160
Estimation du nombre de billets à vendre pour maximiser le profit :  135


2. Déterminer une valeur approchée du profit moyen par avion lorsque le nombre de billets vendus est n=130.

**Lorsque n = 130, le profit moyen est d’environ 12170 euros par avion.**

3. Réaliser plusieurs tests et conjecturer le nombre idéal de billets à proposer à la vente.




** On peut estimer que le nombre idéal de billets à vendre se situe entre 133 et 138.**

4. Déterminer le nombre de billets à vendre pour maximiser le profit, sachant que le maximum de billets à vendre est 140.


 **En faisant des tests plus poussés, on trouve que le nombre idéal de billets à vendre
est 135.**

bONUS1:On suppose que le nombre de billets vendus maximise le profit. Calculer ce profit.

On suppose dans cette question que n = 135 billets ont été vendus. Soit G la
variable aléatoire donnant le profit de la compagnie aérienne. 

On a alors E(X) =0.91 × 135 = 122.85. 

Si X $<=$ 124, alors G = 100X + 30(n − X) = 70X + 4 050.

D’après le chapitre 13, on a alors E(G) = 70E(X) + 4 050 = 12 649; 50.

Si X > 124, certains clients ne peuvent pas embarquer donc G = 100×124+ 30(n−
X) − 150(X − 124) = −180X + 35 050.

On a alors E(G) = −180E(X) + 35 050 = 12 937.

bONUS2. En utilisant la méthode de surréservation décrite, quel est le gain réalisé en comparaison à une compagnie qui ne pratiquerait pas la surréservation mais qui ne rembourserait pas les billets ?

Une entreprise qui ne pratiquerait pas la surréservation vendrait 124 billets. 

Comme les billets ne sont pas remboursés, chaque client rapporte, présent ou non, 100 euros.

Le profit associé à chaque vol est alors 12 400 euros.

La pratique de la surréservation permet à la compagnie de faire un profit légèrement supérieur, et permet de soigner l’image de la société, en permettant des remboursements aux clients non présents,
et en compensant financièrement les clients ne pouvant embarquer.